plotly/tools.py

```diff
@@ -1443,30 +1443,320 @@ class FigureFactory(object):
-        import plotly.graph_objs as go
-        import pandas as pd
-        import random as r
-       
-        colors = ["rgb(31, 119, 180)", "rgb(255, 127, 14)",
-                  "rgb(44, 160, 44)", "rgb(214, 39, 40)",
-                  "rgb(148, 103, 189)", "rgb(140, 86, 75)",
-                  "rgb(227, 119, 194)", "rgb(127, 127, 127)",
-                  "rgb(188, 189, 34)", "rgb(23, 190, 207)"]
+        # Insert traces into trace_list
+        for listy in matrix:
+            for listx in matrix:
+                if (listx == listy) and (diag == "Histogram"):
+                    trace = graph_objs.Histogram(
+                        x = listx,
+                        showlegend = False
+                        )
+                elif (listx == listy) and (diag == "box"):
+                    trace = graph_objs.Box(
+                        y = listx,
+                        name = None,
+                        showlegend = False
+                        )
+                else:
+                    # Add Jitter
+                    if not isinstance(listx[0], basestring):
+                        spanx = abs(max(listx) - min(listx))
+                        for j in range(len(listx)):
+                            listx[j] = listx[j] + (spanx/2)*jitter*gauss(0,1)
+                    if not isinstance(listy[0], basestring):
+                        spany = abs(max(listy) - min(listy))
+                        for j in range(len(listy)):
+                            listy[j] = listy[j] + (spany/2)*jitter*gauss(0,1)
+
+                    trace = graph_objs.Scatter(
+                            x = listx,
+                            y = listy,
+                            mode = "markers",
+                            marker = dict(
+                                symbol = symbol,
+                                size = size),
+                            showlegend = False
+                            )
+                trace_list.append(trace)
+
+        # Create list of index values for the axes
+        indicies = range(dim)
+        indicies.remove(0)
+        indicies.append(dim)
+        j = 0
+
+        for y_index in indicies:
+            for x_index in indicies:
+                fig.append_trace(trace_list[j], y_index, x_index)
+                j += 1
+        # Insert headers into the figure
+        for j in range(dim):
+            xaxis_place = "xaxis" + str(dim*dim - dim+1+j)
+            fig['layout'][xaxis_place].update(title = headers[j])
+        for j in range(dim):
+            yaxis_place = "yaxis" + str(1 + dim*j)
+            fig['layout'][yaxis_place].update(title = headers[j])
+
+        # Set height and width, if not already set by user
+        if (height is None):
+            if dim <= 4:
+                height = 985
+            else:
+                height = 1231
+        if (width is None):
+            if dim <= 4:
+                width = 985
+            else:
+                width = 1231
```

Comment:
Can we create module-scoped constants for the width and height? Or, something like that to remove the need for magic :tophat: numbers here?